lib/views/multi-file-patch-view.js

```diff
@@ -304,13 +303,47 @@ export default class FilePatchView extends React.Component {
     );
   }
 
-  renderExecutableModeChangeMeta() {
-    if (!this.props.filePatch.didChangeExecutableMode()) {
+  renderFilePatchDecorations = filePatch => {
+    return (
+      <Fragment key={filePatch.getPath()}>
+        <Marker invalidate="never" bufferRange={filePatch.getStartRange()}>
+          <Decoration type="block" className="github-FilePatchView-controlBlock">
+            <Fragment>
+
+              <FilePatchHeaderView
+                itemType={this.props.itemType}
+                relPath={filePatch.getPath()}
+                stagingStatus={this.props.stagingStatus}
+                isPartiallyStaged={this.props.isPartiallyStaged}
+                hasHunks={filePatch.getHunks().length > 0}
+                hasUndoHistory={this.props.hasUndoHistory}
+                hasMultipleFileSelections={this.props.hasMultipleFileSelections}
+
+                tooltips={this.props.tooltips}
+
+                undoLastDiscard={() => this.undoLastDiscardFromButton(filePatch)}
+                diveIntoMirrorPatch={() => this.props.diveIntoMirrorPatch(filePatch)}
+                openFile={() => this.didOpenFile({selectedFilePatch: filePatch})}
+                toggleFile={() => this.props.toggleFile(filePatch)}
+              />
+              {this.renderExecutableModeChangeMeta(filePatch)}
+              {this.renderSymlinkChangeMeta(filePatch)}
```

Comment:
@smashwilson: I moved this function call from line 312 to here, and now the integration tests in `file-patch.test.js` are failing.  😱 

The tests pass when I move the rendering back to the way it was, so I'm confident that this is the problem, but I don't understand why.

Katrina and I tried to figure it out today and failed.  Would you be willing to take a look?